tests/func-tests/wasp_agent_test.go

```diff
@@ -0,0 +1,158 @@
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	securityv1 "github.com/openshift/api/security/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+)
+
+const (
+	waspAgentComponentLabel     = "app.kubernetes.io/component=wasp-agent"
+	dsName                      = "wasp-agent"
+	saName                      = "wasp"
+	sccName                     = "wasp"
+	clusterRoleName             = "wasp-cluster"
+	clusterRoleBindingName      = "wasp-cluster"
+	setMemoryOvercommitTemplate = `[{"op": "replace", "path": "/spec/higherWorkloadDensity/memoryOvercommitPercentage", "value": %d}]`
+)
+
+var _ = Describe("Test wasp-agent", Label(tests.OpenshiftLabel, "wasp-agent"), Serial, Ordered, func() {
+	tests.FlagParse()
+
+	var (
+		cli                   client.Client
+		originalHigherDensity *hcov1beta1.HigherWorkloadDensityConfiguration
+	)
+
+	BeforeAll(func(ctx context.Context) {
+		cli = tests.GetControllerRuntimeClient()
+		tests.FailIfNotOpenShift(ctx, cli, "wasp-agent")
+		originalHco := tests.GetHCO(ctx, cli)
+		if originalHco.Spec.HigherWorkloadDensity != nil {
+			originalHigherDensity = &hcov1beta1.HigherWorkloadDensityConfiguration{
+				MemoryOvercommitPercentage: originalHco.Spec.HigherWorkloadDensity.MemoryOvercommitPercentage,
+			}
```

Comment:
Should we patch the HyperConverged CR with these changes?

Reply:
@nunnatsa My assumption was that we need to restore the HCO CR original values as they were before this test, because I don't know what is the "baseline" that is being used for the e2e.

Reply:
but the the code populate the field... so is it the original state?